src/plugins/kibana/server/routes/api/ingest/register_post.js

```diff
@@ -73,8 +74,13 @@ export function registerPost(server) {
       delete indexPattern.id;
 
       const mappings = createMappingsFromPatternFields(indexPattern.fields);
-      indexPattern.fields = initDefaultFieldProps(indexPattern.fields);
 
+      const metaFields = _.get(config, 'metaFields.userValue', config.metaFields.value);
```

Comment:
Consumers of the `uiSettings` shouldn't need to know internals like this. I suggest we add a `.get(key)` method like:

`key => getAll().then(config => _.get(config, `${key}.userValue`, config[key].value))`